From d53ade3f03ba3557fd50fb38990d605f4ae7f8f1 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Sat, 25 Dec 2021 18:04:18 -0800
Subject: [PATCH 2/19] int to string

f2c does not handle implicit casts of function arguments correctly. The msg
argument of `xerrwv` is defined to be an `int *`, and then implicitly cast
from a string at the call site. This doesn't work correctly.

We redefine the type of the first argument to be string to fix the problem.
---
 scipy/integrate/odepack/xerrwv.f | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/scipy/integrate/odepack/xerrwv.f b/scipy/integrate/odepack/xerrwv.f
index 7e180e4f8..b940bb702 100644
--- a/scipy/integrate/odepack/xerrwv.f
+++ b/scipy/integrate/odepack/xerrwv.f
@@ -1,5 +1,6 @@
       subroutine xerrwv (msg, nmes, nerr, level, ni, i1, i2, nr, r1, r2)
-      integer msg, nmes, nerr, level, ni, i1, i2, nr,
+      character  msg*1
+      integer nmes, nerr, level, ni, i1, i2, nr,
      1   i, lun, lunit, mesflg, ncpw, nch, nwds
       double precision r1, r2
       dimension msg(nmes)
-- 
2.34.1

